### PR TITLE
Fixing bug and correcting error messages

### PR DESCRIPTION
**Pull request by MbolotSuse**

### CODE REVIEW

**Comment by MbolotSuse** [comment]
Related to rancher/rancher#37920 and rancher/rancher#37921

**Comment by MbolotSuse** [comment]
There are two sets of messages produced by the adapter - one present in the compliance banner and one present in the support config.

In the error case, you can find these messages here (order is message in configmap/supportconfig, message in user notification):

https://github.com/rancher/csp-adapter/pull/11/files#diff-67c0c944a6ee6e895e8fbe4319b7e328fa36f4fcd798197001ae3ddf8f41ca52R58

In the pass/fail case, you can find the message in the configmap/supportconfig here:

https://github.com/rancher/csp-adapter/pull/11/files#diff-67c0c944a6ee6e895e8fbe4319b7e328fa36f4fcd798197001ae3ddf8f41ca52R151

In the fail case, you can find the message for the UserNotifcation/banner here:

https://github.com/rancher/csp-adapter/pull/11/files#diff-67c0c944a6ee6e895e8fbe4319b7e328fa36f4fcd798197001ae3ddf8f41ca52R149

File: pkg/manager/aws.go [hunk 1]
```diff
@@ -143,10 +146,11 @@ func (m *AWS) runComplianceCheck(ctx context.Context) error {
 	if currentCheckoutInfo.EntitledLicenses == requiredLicenses {
 		statusMessage = "Rancher server has the required amount of licenses"
 	} else {
-		statusMessage = fmt.Sprintf("server is not in compliance, wanted %d, but got %d", requiredLicenses, currentCheckoutInfo.EntitledLicenses)
+		statusMessage = fmt.Sprintf("Server is not in compliance with license terms. At least %d more licens(es) are required in %s to become compliant.", requiredLicenses-currentCheckoutInfo.EntitledLicenses, awsCSP)
```
Comment (cbron) [inline pkg/manager/aws.go]: AWS should be capitalized. "License terms" is a bit formal, and "server" is too broad. How about: 
```suggestion
		statusMessage = fmt.Sprintf("You have exceeded your licensed node count. At least %d more licens(es) are required in %s to become compliant.", requiredLicenses-currentCheckoutInfo.EntitledLicenses, awsCSP)
```

File: pkg/manager/aws.go [hunk 2]
```diff
@@ -143,10 +146,11 @@ func (m *AWS) runComplianceCheck(ctx context.Context) error {
...
 	}
+	configMessage := fmt.Sprintf("Rancher server required %d licens(es) and was able to check out %d licens(es)", requiredLicenses, currentCheckoutInfo.EntitledLicenses)
```
Comment (cbron) [inline pkg/manager/aws.go:154]: can you comment here or in updateAdapterOutput that this goes into the support config (if it doesn't already)

File: pkg/manager/aws.go [hunk 3]
```diff
@@ -55,7 +55,7 @@ func (m *AWS) start(ctx context.Context, errs chan<- error) {
 	for range ticker(ctx, managerInterval) {
 		err := m.runComplianceCheck(ctx)
 		if err != nil {
-			updError := m.updateAdapterOutput(false, fmt.Sprintf("unable to run compliance check with error: %v", err), "unable to run adapter, check adapter logs")
+			updError := m.updateAdapterOutput(false, fmt.Sprintf("unable to run compliance check with error: %v", err), "Unable to run the csp adapter, please check the adapter logs")
```
Comment (cbron) [inline pkg/manager/aws.go]: @wjimenez5271 I'm thinking we should prepend every banner message with `AWS Marketplace: {message}` or similar, so people understand the context, wdyt ?

File: pkg/manager/aws.go (hunk 2 above)
Comment (MbolotSuse) [reply pkg/manager/aws.go:154]: Added a comment in the docstring indicating what each parameter is used for. Let me know if you want another comment elsewhere.

File: pkg/manager/aws.go (hunk 1 above)
Comment (MbolotSuse) [reply pkg/manager/aws.go]: Change made, good suggestion.

File: pkg/manager/aws.go (hunk 3 above)
Comment (wjimenez5271) [reply pkg/manager/aws.go]: Agreed. I would suggest the prepended text be `AWS Marketplace Adapter:`

Comment (cbron) [reply pkg/manager/aws.go]: Cool. Let's change this message to just be "Unable to run the adapter...", and drop the "CSP" phrase from UX messages.

Comment (MbolotSuse) [reply pkg/manager/aws.go]: Ok - to summarize the changes, I will:
- Add the Prefix "AWS Marketplace Adapter" to all UI facing messges
- Remove any reference to the words "csp"

Let me know if I missed something

**Comment by MbolotSuse** [comment]
The adapter produces two messages - one which goes in the support config bundle, and one which is eventually displayed to the user. Below is an overview of what they will say in various scenarios.

## In compliance

### Support Config Message

"Rancher server required x licens(es) and was able to check out y licens(es)", where x is the number of licenses that we need to check out due to this server's node count and y is the number of licenses that we were able to check out.

For example, if we have 20 nodes, this would read:

"Rancher server required 1 licens(es) and was able to check out 1 licens(es)"

### User Notification/ Banner
N/A - No notification is made if we are in compliance

## Out of Compliance

### Support Config Message

"Rancher server required x licens(es) and was able to check out y licens(es)", where x is the number of licenses that we need to check out due to this server's node count and y is the number of licenses that we were able to check out.

For example, if we have 40 nodes, but had only purchased 1 license this would read:

"Rancher server required 2 licens(es) and was able to check out 1 licens(es)"


### User Notification/ Banner

"AWS Marketplace Adapter: You have exceeded your licensed node count. At least x more licens(es) are required in AWS to become compliant.", where x is the number of licenses that you need to purchase to become compliant. 

For example, if we have 40 nodes, but had only purchased 1 license this would read:

"AWS Marketplace Adapter: You have exceeded your licensed node count. At least 1 more licens(es) are required in AWS to become compliant."

## Adapter Was unable to run due to an error

### Support Config Message

"unable to run compliance check with error: x", where x is the error that the CSP adapter encountered when attempting to run.

For example, if the adapter got a "unable to get rancher license, err: permission denied", this would say

"unable to run compliance check with error: unable to get rancher license, err: permission denied"

### User Notification/ Banner

"AWS Marketplace Adapter: Unable to run the adapter, please check the adapter logs."

File: pkg/manager/aws.go (hunk 3 above)
Comment (MbolotSuse) [reply pkg/manager/aws.go]: I've added the prefix and removed reference to the CSP part of this from the user facing messages. Thank you for the feedback. The PR also has a new comment outlining the desired messages and the scenarios that they are present in - please let me know if either of you have a suggestion on changes that I should make to the messages.

**Review by cbron — APPROVED** [review]
lgtm

**Pull request merged**